### ISSUE TITLE
Using pyfakefs breaks vscode's debugger

### ISSUE THREAD

**Issue by Aran-Fey** [pytest-dev/pyfakefs#1083]
Using pyfakefs leads to a variety of issues with vscode's debugger, like breakpoints not working and duplicate file tabs being opened. I made a bug report [here](https://github.com/microsoft/debugpy/issues/1709), but the folks at debugpy have decided it's not their responsibility to fix it. Can you guys take care of it?

**Comment by Aran-Fey** [comment]
This may only be an issue if `PatchMode.AUTO` is used, now that I think about it.

**Comment by mrbean-bremen** [comment]
Yeah, `PatchMode.AUTO` may have unwanted side effects and probably won't play nice with a debugger, as that relies on loading code on a low level
If this indeed only happens if patching `open_code`, there is probably not much we can do with reasonable effort. I don't think I want to go down that rabbit hole...

**Comment by mrbean-bremen** [comment]
@Aran-Fey - can you confirm that this only happens with `PatchMode.AUTO`?

**Comment by Aran-Fey** [comment]
Actually, nope. A duplicate tab is opened regardless of the `PatchMode`, and the breakpoints continue to *sometimes* be ignored without any apparent reason.

**Comment by mrbean-bremen** [comment]
Ok, thanks. I'm not using VSCode (I use PyCharm), so it may take a while to understand this. Do you have a way to reproduce this with some minimal example?

**Comment by Aran-Fey** [comment]
This seems to do it:

```python
import typing

def test_demo(fs):
    typing.get_type_hints(int)
```

1. Set a breakpoint at `typing.get_type_hints(int)` -> breakpoint is hit, duplicate file tab is opened
2. Set a breakpoint *inside of* `get_type_hints` -> breakpoint is ignored
3. Remove `fs` fixture: Both issues go away

**Comment by mrbean-bremen** [comment]
Thanks, I will give it a try!

**Comment by mrbean-bremen** [comment]
You are working with the current pyfakefs version (5.7.1), right?

**Comment by mrbean-bremen** [comment]
Ah, and which Python version are you using?

**Comment by mrbean-bremen** [comment]
So far, I have not been able to get sensible behavior even in tests without pyfakefs - I guess I have to better understand the workflow in VSCode first... Will have another go over the weekend.

**Comment by mrbean-bremen** [comment]
Ok, I persuaded VSCode to work for me, and I can reproduce the duplicate tab behavior now (not the ignored breakpoint so far). At the moment, I have no idea where to even look, so no promises...

**Comment by mrbean-bremen** [comment]
@Aran-Fey - can you please check the behavior with the main branch?

**Comment by Aran-Fey** [comment]
The issue with the read-only tabs is fixed. Not sure about the other one, but this is definitely much better than before.
